### PR TITLE
Cw posts

### PR DESCRIPTION
**Pull request by Cwelshofer**
## Tickets
_My PR references these tickets (epic and task tickets)_

1. #130 testing for filter Id, #129 Viewing one post, #122 Viewing all posts.
​

## How do I access this branch?

1. ```git fetch --all```
1. ```git checkout ``` 
1. ```python manage.py runserver``` 

​

​
## Types of changes
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Testing
- [ x] Tested in Postman
- [ ] Tested with Client

``` if necessary, postman code here ```

#130- url: http://localhost:8000/posts?user_id=1 
#130-GET in postman also check of user_id=2
#122- GET in postman check link http://localhost:8000/posts
#129- GET in postman check link http://localhost:8000/posts/1 and http://localhost:8000/posts/2

## Checklist for Reviewer
- [ ] check it out
- [ ] test all edge cases listed and more if you think of them (postman or browser)
- [ ] request changes/approve
- [ ] review code to make sure it fits coding style of project

### CODE REVIEW

**Review by egeehartz — APPROVED** [review]

File: rareapi/views/post.py [hunk 1]
```diff
@@ -1,40 +1,62 @@
...
+        depth = 1
+
+"""Basic Serializer for single post"""
+class PostSerializer(serializers.ModelSerializer):
+    user = PostRareUserSerializer(many=False)
     class Meta:
         model = Post
-        fields = ('id', 'user', 'category', 'title', 'publication_date', 'image_url')
\ No newline at end of file
+        fields = ('id', 'title', 'publication_date', 'content', 'user', 'category')
```
Comment (esthermcmahon) [inline rareapi/views/post.py]: this needs to include fields = ('id', 'user', 'category', 'title', 'publication_date', 'image_url', 'content', 'approved')

**Review by egeehartz — CHANGES_REQUESTED** [review]
GET /posts returns one post (the first one on the list)
GET posts?user_id=2 returns one post (the first post on the list)

GET single post works

**Review by esthermcmahon — APPROVED** [review]
great!

**Review by egeehartz — APPROVED** [review]
/posts gets all posts  👍 
/post/postId gets the correct post 👍 
/post?user_id=2 gets all of user 2's posts 👍
